src/VPolygon.jl

```diff
@@ -553,3 +553,52 @@ function translate(P::VPolygon{N}, v::AbstractVector{N}) where {N<:Real}
                                 "set by a $(length(v))-dimensional vector"
     return VPolygon([x + v for x in vertices_list(P)])
 end
+"""
+    minkowski_sum(P::VPolygon{N}, Q::VPolygon{N}) where {N<:Real}
+
+The Minkowski Sum of two polygon in vertex representation.
+
+### Input
+
+- `P` -- polygon in vertex representation
+- `Q` -- another polygon in vertex representation
+
+### Output
+
+A polygon in vertex representation.
+
+### Algorithm
+
+We treat each edge of the polygons as a vector, attaching them in polar order
+(attaching the tail of the next vector to the head of the previous vector). The
+resulting polygonal chain will be a polygon, which is the Minkowski sum of the given polygons.
+"""
+
+function minkowski_sum(P::VPolygon{N}, Q::VPolygon{N}) where {N<:Real}
+    vlP = vertices_list(P)
+    vlQ = vertices_list(Q)
+    mP = length(vlP);
+    mQ = length(vlQ);
+    i = 1
+    k = 1
+    j = 1
+    R = Vector{Vector{Float64}}(undef, mP+mQ)
+    fill!(R, N[0, 0])
+    while i <= size(R, 1)
+        OP = vlP[(k%mP+1)]-vlP[(k-1)%mP+1]
+        OQ = vlQ[(j%mQ+1)]-vlQ[(j-1)%mQ+1]
+        R[i] = vlP[(k-1) % mP+1] + vlQ[(j-1) % mP+1]
+        is_rightTurn = LazySets.right_turn(OP, OQ, N[0, 0])
+        if is_rightTurn > 0
+            k += 1
+        elseif (is_rightTurn < 0)
+            j += 1
+        else
+            pop!(R)
```

Comment:
is this final `else` there to cover the corner case that the line segments are parallel?

Reply:
Yes, when two line segments are parallel and are in the same direction in the polygons, between the resulting edges there is not a vertex. If it wasn't for the pop!(R), it would be a vertex at the origin.